lib/class-wp-block-type.php

```diff
@@ -149,6 +149,16 @@ public function prepare_attributes_for_render( $attributes ) {
 			$prepared_attributes[ $attribute_name ] = $value;
 		}
 
+		/**
+		* Filter to allow plugins to override block attributes
+		* 
+		* @since 3.5.0
+		* 
+		* @param array $prepared_attributes block attributes 
+		*/
+
+		$prepared_attributes = apply_filters( 'block_attributes', $prepared_attributes );
```

Comment:
IMO `$this` should be passed to the filter as well. Otherwise one has no idea which block type the attributes are for.

Reply:
Can't it pass the block name instead? What other information plugins would need?